.pre-commit-config.yaml

```diff
@@ -162,3 +162,14 @@ repos:
           |^NOTICES$
           |^proto/streamlit/proto/openmetrics_data_model.proto$
           |\.snap$
+
+  - repo: https://github.com/pycqa/pydocstyle
+    rev: 6.1.1
+    hooks:
+      - id: pydocstyle
+        name: Run pydocstyle
+        args:
+          - --convention=numpy
+          - --add-ignore=D100,D101,D104,D105,D106,D107
+          - --add-select=D417
+          - --match-dir=(lib\/streamlit\/)(?!vendor)(?!watcher)
```

Comment:
If we use the `files`/`exclude` option for pre-commit-hook, this check will only run when we make changes to matching files. 
<img width="1168" alt="Screenshot 2022-11-29 at 16 27 06" src="https://user-images.githubusercontent.com/78743291/204571031-3b67f9a1-965d-4a91-9ed3-21752f45dbb2.png">

Now, this check will be run when we modify any python file.
```
    types: [python]
```
https://github.com/PyCQA/pydocstyle/blob/master/.pre-commit-hooks.yaml#L6